e2e_tests/test_repos_find.py

```diff
@@ -101,9 +102,11 @@ def test_repos_find(configtype, default):
         args = ["repos", "find", "local", tmpdir]
         if not default:
             args += ["--format", configtype]
+            if exclude:
+                args += ["--exclude", exclude]
         cmd = grm(args)
         assert cmd.returncode == 0
-        assert len(cmd.stderr) == 0
+        assert len(cmd.stderr) == (37 if exclude == "^.*/repo2$" and not default else 0)
```

Comment:
I tried to not be too exact when matching error output. I think it's preferable to just check for some words that you *know* must be in the output:

```python
if exclude == "^.*/repo2$":
    assert "[skipped]" in cmd.stderr.lower()
    assert "repo2" in cmd.stderr.lower()
else:
    assert len(cmd.stderr) == 0
```

This makes the tests a bit more robust against small changes in the code. Otherwise, something innocent like adding full stops at the end of the error output would break the tests.